src/state/state.js

```diff
@@ -188,14 +188,14 @@
          * @ignore
          */
         function _renderFrame(time) {
-            // update all game objects
-            me.game.update(time);
-            // render all game objects
-            me.game.draw();
             // schedule the next frame update
             if (_animFrameId !== -1) {
                 _animFrameId = window.requestAnimationFrame(_renderFrame);
             }
+            // update all game objects
+            me.game.update(time);
+            // render all game objects
+            me.game.draw();
```

Comment:
I'm not sure of the validity of this suggestion, though. I've seen no tests to prove that it actually creates a smoother animation. It's probably all based on perception and assumptions.